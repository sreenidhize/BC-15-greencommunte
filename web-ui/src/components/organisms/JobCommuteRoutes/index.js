import { Divider, Grid, Paper } from '@material-ui/core';
import React, {useState} from 'react';
import JobDetailsHeader from '../../molecules/JobDetailsHeader/JobDetailsHeader';
import {makeStyles, Snackbar} from "@material-ui/core";
import ArrowBack from "@material-ui/icons/ArrowBack";
import CommuteRoutesDescription from '../CommuteRoutesDescription/CommuteRoutesDescription';
import CommuteOptions from '../CommuteOptions/CommuteOptions';
import Button from '../../atoms/button/Button';
import {Alert} from "@material-ui/lab";
import { apis } from '../../../utils/resources/resources';
import { headerInfo, jobSavedSuccess, jobAlreadyPresent, jobSavedFailed, failedToRemove } from '../../../utils/constants/constant';


const useStyles = makeStyles((theme) => ({
    root: {
        width: '390px',
    },
    divider: {
        marginTop: '30px',
        marginLeft: 15,
        marginBottom: '30px',
        width: 360,
    },
    backIcon: {
        color: theme.palette.secondary.main,
        paddingRight: theme.spacing(1),
        marginTop: -3.5,
        cursor: 'pointer'
    },
    buttonLabel: {
        fontFamily: theme.typography.fontFamily,
        height: theme.spacing(2),
        fontWeight: '500',
        marginTop: 10,
    },
    JobCommuteRoutes: {
        marginTop: 15,
        marginLeft: 25,
    },
    CommuteOptions: {
        padding: 10,
        marginLeft: 10,
    },
    BackButton: {
        marginTop: -6,
        marginLeft: 25,
    },
    JobHeader: {
        marginTop: 30,
        marginLeft: 25,
        marginBottom: -10,
    },
    Paper: {
        width: 390,
        border: 0,
        borderTop: "solid 1px #e3f3f6",
        height: "calc(100vh - 73px)",
        overflowY: "scroll",
        overflowX: "hidden"
    },
    buttonsStyle: {
        width: "390px",
        display: "flex",
        marginTop: theme.spacing(4),
        marginBottom: 25,
    },
    LastDivider: {
        width: 390,
        marginTop: 10,
        marginBottom: 10,
        color: "#e3f3f6"
    }
}));

const JobCommuteRoutes = (props) => {
    const classes = useStyles();
    const { url, alt, jobTitle, companyName, location, from, to, onBackClick,
        onClickApply, jobId, commuteTypes, cabOptions, showFindJobs } = props;
    const saveOrRemoveButton = showFindJobs ? "Save" : "Remove";
    const [message, setMessage] = useState("Message");
    const jobsUrl = new URL(apis.GET_SAVEDJOBS + jobId);
    const [messageType, setMessageType] = useState("success");
    const savedJobsUrl = new URL(apis.GET_SAVEDJOBS + jobId);
    const [showMessage, setShowMessage] = useState(false);
    const onClickSaveOrRemove = () => {
        if(saveOrRemoveButton === "Save") {
            fetch(jobsUrl, { 
                method: 'POST', 
                headers: {
                    "Content-type": headerInfo
                } 
            }) 
            .then((res) => {
                if (res.status === 200){
                    setMessage(jobSavedSuccess);
                    setMessageType("success");
                } else if (res.status === 409){
                    setMessageType("info");
                    setMessage(jobAlreadyPresent);
                } else {
                    setMessage(jobSavedFailed);
                    setMessageType("error");
                }
                setShowMessage(true);
            });
        }
        else {
            fetch(savedJobsUrl, {
            method: 'DELETE'})
            .then(res => {
                if (res.status === 200){
                    props.itemsRemoved(jobId);
                } else {
                    setMessage(failedToRemove);
                    setMessageType("error");
                    setShowMessage(true);
                }
                setShowMessage(true);
            });

        }
    };
    return (
        <div data-testid="job-commute-routes">
            <Paper className={classes.Paper} elevation={0} variant="outlined" square>
                <Grid container direction="column" className={classes.root}>
                    <Grid item className={classes.JobHeader}>
                        <JobDetailsHeader
                            url={url}
                            alt={alt}
                            jobTitle={jobTitle}
                            location={location}
                            companyName={companyName}
                        />
                    </Grid>
                    <Grid item>
                        <Divider variant="middle" className={classes.divider}></Divider>
                    </Grid>
                    <Grid container direction="row" className={classes.BackButton}>
                        <Grid item>
                            <ArrowBack className={classes.backIcon} onClick={onBackClick}/>
                        </Grid>
                        <Grid item>
                            <span className={classes.buttonLabel}>Commute Routes</span>
                        </Grid>
                    </Grid>
                    <Grid item className={classes.JobCommuteRoutes}>
                        <CommuteRoutesDescription from={from} to={to} />
                    </Grid>
                    <Grid item>
                        <Divider variant="middle" className={classes.divider}></Divider>
                    </Grid>
                    <Grid item className={classes.CommuteOptions}>
                        <CommuteOptions commuteTypes={commuteTypes} cabOptions={cabOptions} selectedCommute="metro" />
                    </Grid>
                    <Grid item>
                        <Divider className={classes.LastDivider}></Divider>
                    </Grid>
                    <Grid item>
                        <Grid container justify="space-around" alignItems="center" className={classes.buttonsStyle} >
                            <Button label={saveOrRemoveButton} type="medium2" onClick={onClickSaveOrRemove} />
                            <Button label="Apply" variant="contained" type="medium2" onClick={onClickApply} />
                        </Grid>
                    </Grid>
                    <Snackbar
                        anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                        open={showMessage}
                        autoHideDuration={3000}
                        onClose={()=> {setShowMessage(false);}}
                    >
                    <Alert severity={messageType} elevation={6} variant="filled">
                        {message}
                    </Alert>
                    </Snackbar>
                </Grid>
            </Paper>
        </div>
    );
};

export default JobCommuteRoutes;