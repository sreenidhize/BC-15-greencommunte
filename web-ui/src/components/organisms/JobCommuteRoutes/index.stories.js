import React from 'react';
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';
import JobCommuteRoutes from '.';

export default {
    title: 'Organisms/Routes',
    component: JobCommuteRoutes
};

const Template = (args) => <ThemeProvider theme={theme}><JobCommuteRoutes {...args} /></ThemeProvider>;

export const Primary = Template.bind({});
Primary.args = {
    url: "https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg",
    alt: "BMW logo",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location:"Hitech City, Telangana",
    isPrimary: false,
    from: "East Marredpally, E Marredpally",
    to: "Hitech City, Telangana, Secunderabad",
    onClickSave: () => {alert("save click");},
    onClickApply: () => {alert("apply click");},
    commuteTypes: [
        {
            type: 'bus',
            fare: '15$',
            time: '20 mins',
            description: "Catch a bus 38X at 3.42 PM to Secunderabad bus stand.",
            lateStatus: '2 mins',
            isLate: true
        },
        {
            type: 'walk',
            time: '5 mins',
            description: "Head north on St. Johns road."
        },
        {
            type: 'metro',
            fare: '56$',
            time: '38 mins',
            description: "Catch a blue line metro towards Raidurg.",
            isLate: false
        }
    ]
};