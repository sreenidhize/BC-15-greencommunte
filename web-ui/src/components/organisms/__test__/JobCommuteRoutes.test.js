import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import JobCommuteRoutes from "../JobCommuteRoutes";

const url = "https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg";
const alt = "BMW logo";
const commuteTypes = [
  {
    type: "bus",
    fare: "15$",
    time: "20 mins",
    description: "Catch a bus 38X at 3.42 PM to Secunderabad bus stand.",
    lateStatus: "2 mins",
    isLate: true,
  },
  {
    type: "walk",
    time: "5 mins",
    description: "Head north on St. Johns road.",
  },
  {
    type: "metro",
    fare: "56$",
    time: "38 mins",
    description: "Catch a blue line metro towards Raidurg.",
    isLate: false,
  },
];

describe("Job Commute Routes placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <JobCommuteRoutes url={url} alt={alt} commuteTypes={commuteTypes} />
    );
    expect(container).toMatchSnapshot();
  });

  test("test Job Commute Routes render", () => {
    const { getByTestId } = render(
      <JobCommuteRoutes url={url} alt={alt} commuteTypes={commuteTypes} />
    );
    expect(getByTestId("job-commute-routes")).toBeInTheDocument();
  });
});
