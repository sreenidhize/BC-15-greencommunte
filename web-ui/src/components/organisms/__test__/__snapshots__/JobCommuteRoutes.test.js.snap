// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Commute Routes placeholder test Matches the snapshot 1`] = `
<div>
  <div
    data-testid="job-commute-routes"
  >
    <div
      class="MuiPaper-root makeStyles-Paper-9 MuiPaper-outlined"
    >
      <div
        class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-direction-xs-column"
      >
        <div
          class="MuiGrid-root makeStyles-JobHeader-8 MuiGrid-item"
        >
          <div
            class="MuiGrid-root makeStyles-root-14"
            data-testid="jobDetailsHeader-placeholder"
          >
            <div
              data-testid="img-placeholder"
            >
              <img
                alt="BMW logo"
                class="makeStyles-secondary-16"
                src="https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg"
              />
            </div>
            <div
              class="MuiGrid-root makeStyles-columnMargin-13 MuiGrid-container MuiGrid-direction-xs-column"
            >
              <h4
                class="MuiTypography-root makeStyles-primary-17 MuiTypography-h4"
                data-testid="cardLabel-placeholder"
              />
              <h4
                class="MuiTypography-root makeStyles-secondary-18 MuiTypography-h4"
                data-testid="cardLabel-placeholder"
              />
              <h4
                class="MuiTypography-root makeStyles-secondary-18 MuiTypography-h4"
                data-testid="cardLabel-placeholder"
              />
              
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <hr
            class="MuiDivider-root makeStyles-divider-2 MuiDivider-middle"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-BackButton-7 MuiGrid-container"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-backIcon-3"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <span
              class="makeStyles-buttonLabel-4"
            >
              Commute Routes
            </span>
          </div>
        </div>
        <div
          class="MuiGrid-root makeStyles-JobCommuteRoutes-5 MuiGrid-item"
        >
          <div
            data-testid="commuteRoutesDescription-placeholder"
          >
            <img
              alt="Commute Map"
              height="135px"
              src="https://i.stack.imgur.com/8kOXQ.jpg"
              width="330px"
            />
            <div
              class="makeStyles-container-19"
            >
              <div
                class="makeStyles-location-20"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-22"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="8"
                  />
                </svg>
                <h4
                  class="MuiTypography-root makeStyles-font-21 makeStyles-typoMargin-23 MuiTypography-h4"
                >
                   
                  Hyderabad, Telangana, India
                   
                </h4>
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-22 makeStyles-closeIcon-24"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="makeStyles-container-19"
            >
              <div
                class="makeStyles-location-20"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root makeStyles-icon-22"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9z"
                  />
                  <circle
                    cx="12"
                    cy="9"
                    r="2.5"
                  />
                </svg>
                <h4
                  class="MuiTypography-root makeStyles-font-21 makeStyles-typoMargin-23 MuiTypography-h4"
                >
                   
                  Hyderabad, Telangana, India
                   
                </h4>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <hr
            class="MuiDivider-root makeStyles-divider-2 MuiDivider-middle"
          />
        </div>
        <div
          class="MuiGrid-root makeStyles-CommuteOptions-6 MuiGrid-item"
        >
          <div
            class="MuiGrid-root makeStyles-root-25 MuiGrid-container MuiGrid-direction-xs-column"
            data-testid="commute-options"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
            >
              <h4
                class="MuiTypography-root makeStyles-optionsLabel-26 MuiTypography-h4"
              >
                Your Options
              </h4>
              <div
                data-testid="commuteRoutes-placeholder"
              >
                 
                <div
                  class="makeStyles-flexDisplay-31"
                >
                  <div
                    class=""
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarSelected-36 MuiAvatar-colorDefault"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-root-33 makeStyles-iconSelected-35"
                        data-testid="metro"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <circle
                          cx="8.5"
                          cy="14.5"
                          r="1.5"
                        />
                        <circle
                          cx="15.5"
                          cy="14.5"
                          r="1.5"
                        />
                        <path
                          d="M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2l2-2h4l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-4-4-8-4zm0 2c3.51 0 4.96.48 5.57 1H6.43c.61-.52 2.06-1 5.57-1zM6 7h5v3H6V7zm12 8.5c0 .83-.67 1.5-1.5 1.5h-9c-.83 0-1.5-.67-1.5-1.5V12h12v3.5zm0-5.5h-5V7h5v3z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class=""
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-37 MuiAvatar-colorDefault"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-root-33 makeStyles-iconUnselected-34"
                        data-testid="bus"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M4 16c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4s-8 .5-8 4v10zm3.5 1c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm9 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6H6V6h12v5z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class=""
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-37 MuiAvatar-colorDefault"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-root-33 makeStyles-iconUnselected-34"
                        data-testid="car"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M18.92 5.01C18.72 4.42 18.16 4 17.5 4h-11c-.66 0-1.21.42-1.42 1.01L3 11v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.85 6h10.29l1.04 3H5.81l1.04-3zM19 16H5v-4.66l.12-.34h13.77l.11.34V16z"
                        />
                        <circle
                          cx="7.5"
                          cy="13.5"
                          r="1.5"
                        />
                        <circle
                          cx="16.5"
                          cy="13.5"
                          r="1.5"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class=""
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circle makeStyles-avatarUnselected-37 MuiAvatar-colorDefault"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root makeStyles-root-33 makeStyles-iconUnselected-34"
                        data-testid="bike"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM5 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5zm5.8-10l2.4-2.4.8.8c1.3 1.3 3 2.1 5.1 2.1V9c-1.5 0-2.7-.6-3.6-1.5l-1.9-1.9c-.5-.4-1-.6-1.6-.6s-1.1.2-1.4.6L7.8 8.4c-.4.4-.6.9-.6 1.4 0 .6.2 1.1.6 1.4L11 14v5h2v-6.2l-2.2-2.3zM19 12c-2.8 0-5 2.2-5 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 8.5c-1.9 0-3.5-1.6-3.5-3.5s1.6-3.5 3.5-3.5 3.5 1.6 3.5 3.5-1.6 3.5-3.5 3.5z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-space-between"
            >
              <h4
                class="MuiTypography-root makeStyles-routeLabel-27 MuiTypography-h4"
              >
                Maredpally - Hitech City
              </h4>
              <h6
                class="MuiTypography-root makeStyles-routeDetails-29 MuiTypography-subtitle2"
              >
                $50 • 56 mins
              </h6>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="makeStyles-transitType-39"
                data-testid="transit-type"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-38"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M4 16c0 .88.39 1.67 1 2.22V20c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h8v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1.78c.61-.55 1-1.34 1-2.22V6c0-3.5-3.58-4-8-4s-8 .5-8 4v10zm3.5 1c-.83 0-1.5-.67-1.5-1.5S6.67 14 7.5 14s1.5.67 1.5 1.5S8.33 17 7.5 17zm9 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zm1.5-6H6V6h12v5z"
                    />
                  </svg>
                </div>
                <div
                  class="makeStyles-descDiv-40"
                >
                  <h6
                    class="MuiTypography-root makeStyles-transitDesc-43 MuiTypography-subtitle2"
                  >
                    Catch a bus 38X at 3.42 PM to Secunderabad bus stand.
                    <span
                      class="makeStyles-lateStatus-41"
                    >
                      2 mins
                    </span>
                  </h6>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    15$
                     • 
                    20 mins
                  </h6>
                </div>
              </div>
              <div
                class="makeStyles-transitType-39"
                data-testid="transit-type"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-38"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13.5 5.5c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zM9.8 8.9L7 23h2.1l1.8-8 2.1 2v6h2v-7.5l-2.1-2 .6-3C14.8 12 16.8 13 19 13v-2c-1.9 0-3.5-1-4.3-2.4l-1-1.6c-.4-.6-1-1-1.7-1-.3 0-.5.1-.8.1L6 8.3V13h2V9.6l1.8-.7"
                    />
                  </svg>
                </div>
                <div
                  class="makeStyles-descDiv-40"
                >
                  <h6
                    class="MuiTypography-root makeStyles-transitDesc-43 MuiTypography-subtitle2"
                  >
                    Head north on St. Johns road.
                    <span>
                       
                      5 mins
                    </span>
                  </h6>
                </div>
              </div>
              <div
                class="makeStyles-transitType-39"
                data-testid="transit-type"
              >
                <div>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root makeStyles-root-38"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <circle
                      cx="8.5"
                      cy="14.5"
                      r="1.5"
                    />
                    <circle
                      cx="15.5"
                      cy="14.5"
                      r="1.5"
                    />
                    <path
                      d="M12 2c-4 0-8 .5-8 4v9.5C4 17.43 5.57 19 7.5 19L6 20.5v.5h2l2-2h4l2 2h2v-.5L16.5 19c1.93 0 3.5-1.57 3.5-3.5V6c0-3.5-4-4-8-4zm0 2c3.51 0 4.96.48 5.57 1H6.43c.61-.52 2.06-1 5.57-1zM6 7h5v3H6V7zm12 8.5c0 .83-.67 1.5-1.5 1.5h-9c-.83 0-1.5-.67-1.5-1.5V12h12v3.5zm0-5.5h-5V7h5v3z"
                    />
                  </svg>
                </div>
                <div
                  class="makeStyles-descDiv-40"
                >
                  <h6
                    class="MuiTypography-root makeStyles-transitDesc-43 MuiTypography-subtitle2"
                  >
                    Catch a blue line metro towards Raidurg.
                    <span
                      class="makeStyles-onTimeStatus-42"
                    >
                      On time
                    </span>
                  </h6>
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle2"
                  >
                    56$
                     • 
                    38 mins
                  </h6>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <h6
                class="MuiTypography-root makeStyles-googleMapsLabel-28 MuiTypography-h6 MuiTypography-colorPrimary"
              >
                View in google maps
              </h6>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <hr
            class="MuiDivider-root makeStyles-LastDivider-11"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiGrid-root makeStyles-buttonsStyle-10 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-medium2-46 makeStyles-common-48 MuiButton-outlinedPrimary"
              data-testid="button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Remove
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-medium2-46 makeStyles-common-48 MuiButton-containedPrimary"
              data-testid="button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Apply
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
